### PR TITLE
Some additonal changes to ChainIndexer

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

**Comment by MithrilMan** [comment]
indexer should do the same thing as Get methods, so lock and tryget.

and you can inline Get methods `out` argument
```
ChainedHeader result;
this.blocksById.TryGetValue(id, out result);
```
become this
```
this.blocksById.TryGetValue(id, ChainedHeader out result);
```

**Comment by quantumagi** [comment]
Unit tests on Windows are showing success here but are actually not running:

image

File: src/NBitcoin/ChainIndexer.cs [hunk 1]
```diff
@@ -66,9 +66,9 @@ public void Initialize(ChainedHeader chainedHeader)
                     this.blocksById.Add(iterator.HashBlock, iterator);
                     this.blocksByHeight.Add(iterator.Height, iterator);
 
-                    if (chainedHeader.Height == 0)
+                    if (iterator.Height == 0)
                     {
-                        if (chainedHeader.HashBlock != iterator.HashBlock)
+                        if (this.Network.GenesisHash != iterator.HashBlock)
```
Comment (quantumagi) [inline src/NBitcoin/ChainIndexer.cs]: The `ChainIndexer` is not always used with a first block that matches the genesis block:

```
        private void CanVerifySequenceLockCore(Sequence[] sequences, int[] prevHeights, int currentHeight, DateTimeOffset first, bool expected, SequenceLock expectedLock)
        {
            Network network = this.stratisMain;
            BlockHeader blockHeader = network.Consensus.ConsensusFactory.CreateBlockHeader();
            blockHeader.BlockTime = first;

            var chain = new ChainIndexer(network, new ChainedHeader(blockHeader, blockHeader.GetHash(), 0));
```

That means these tests will have to change or revert this.

**Comment by quantumagi** [comment]
Looks good otherwise. We may want to consider separate PRs for extensive renaming though, where that is all that the PR does.

File: src/NBitcoin/ChainIndexer.cs (hunk 1 above)
Comment (dangershony) [reply src/NBitcoin/ChainIndexer.cs]: Yeah we probably need to change the test, I will have a look.

**Comment by dangershony** [comment]
>We may want to consider separate PRs (or commits) for extensive renaming though.
Totally agree, it will make it much easier to review. I will pay attention next time.

**Comment by fassadlr** [comment]
@dangershony can you check why the unit tests did not run in the build before merging?

**Comment by fassadlr** [comment]
@dangershony I fixed the CI build to rerun the unit tests. Just running our windows build again now.

**Pull request merged**